### PR TITLE
Beam in Docker

### PR DESCRIPTION
**Pull request by REASY**
Uses `docker-java-application` plugin for Gradle to create docker image

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3063)

### CODE REVIEW

**Comment by REASY** [comment]
test!

**Comment by REASY** [comment]
test!

File: build.gradle [hunk 1]
```diff
@@ -598,29 +599,65 @@ docker {
...
+
+// Some examples of manual docker file creation can be found https://github.com/bmuschko/gradle-docker-plugin/blob/3736ff644bed1cbfeed6fc7f844a5513c4208bc2/src/functTest/groovy/com/bmuschko/gradle/docker/DockerWorkflowFunctionalTest.groovy#L24
+// Also here: https://bmuschko.github.io/gradle-docker-plugin/#modifying_instructions_of_a_dockerfile_task
+task createDockerfile(type: Dockerfile, dependsOn: dockerSyncBuildContext) {
+    from 'beammodel/miniconda3'
+    label(['maintainer': 'LBNL Beam Team'])
     instruction 'RUN apt-get update && apt-get -y install python3 python3-pip python3-rtree libproj-dev proj-data proj-bin libgeos-dev npm'
     instruction 'RUN pip3 install --upgrade setuptools'
     instruction 'RUN pip3 install numpy pandas cython sklearn matplotlib mapclassify descartes geopandas contextily plotly collections-extended psutil requests'
-    addFile 'src/main/python/events_analysis/', 'src/main/python/events_analysis/'
```
Comment (JustinPihony) [inline build.gradle:613]: Why remove the events_analysis?

File: build.gradle [hunk 2]
```diff
@@ -598,29 +599,65 @@ docker {
...
+    copyFile("test/input/beamville", "test/input/beamville")
+    copyFile("test/input/sf-light", "test/input/sf-light")
+
+    addFile 'src/main/python', 'src/main/python'
+    addFile 'entrypoint.sh', 'entrypoint.sh'
+    runCommand('chmod +x /app/entrypoint.sh')
+    entryPoint('/app/entrypoint.sh')
+}
+
+task buildImage(type: DockerBuildImage, dependsOn: createDockerfile) {
```
Comment (JustinPihony) [inline build.gradle:634]: Why not just use `createDockerFile`? This doesn't do anything afaict other than piggyback.

File: src/main/scala/beam/sim/BeamHelper.scala [hunk 3]
```diff
@@ -525,7 +532,7 @@ trait BeamHelper extends LazyLogging {
     }
     logger.warn(logStart)
 
-    val injector: inject.Injector = buildInjector(config, beamExecutionConfig.beamConfig, scenario, beamScenario)
+    val injector = buildInjector(config, beamExecutionConfig.beamConfig, scenario, beamScenario)
```
Comment (JustinPihony) [inline src/main/scala/beam/sim/BeamHelper.scala:535]: Why remove annotation - this helps compile time, so if there we should keep there.

**Review by JustinPihony — APPROVED** [review]
Minor items of clarification more than anything

**Comment by JustinPihony** [comment]
test now!

File: build.gradle (hunk 1 above)
Comment (REASY) [reply build.gradle:613]: @JustinPihony don't remember the reason - I was play a bit to make it work first. Reverting that change.

File: build.gradle (hunk 2 above)
Comment (REASY) [reply build.gradle:634]: Due to specifics of the plugin it has to have those two methods (https://bmuschko.github.io/gradle-docker-plugin/#creating_a_dockerfile_and_building_an_image) (as you can see input parameter `type` of those two methods are different: Dockerfile and DockerBuildImage)

**Comment by REASY** [comment]
test now!

**Pull request merged**